extensions/emmet/src/test/wrapWithAbbreviation.test.ts

```diff
@@ -356,6 +356,54 @@ suite('Tests for Wrap with Abbreviations', () => {
 			});
 		});
 	});
+
+	test('Wrap maintains href', () => {
+		const contents = `
+			www.google.com
+		`;
+		const expectedContents = `
+			<a href="http://www.google.com">www.google.com</a>
+		`;
```

Comment:
Update this to 
```
        const contents = `
			www.google.com
            https://github.com
		`;
		const expectedContents = `
			<a href="http://www.google.com">www.google.com</a>
			<a href="https://github.com">https://github.com</a>
		`;
```

And `editor.selections = [new Selection(1, 1, 1, 1), new Selection(2, 1, 2, 1)];`

The test will fail

This is because the [same snippet is applied to all cursors](https://github.com/Microsoft/vscode/blob/29fbb91552b3dbd02093cd9157d5d95219406cb6/extensions/emmet/src/abbreviationActions.ts#L561-L563).

We will have to run `expandAbbr` for all cursors, check if the output is same for all. And then decide whether to call `editor.insertSnippet` for each or just once

Reply:
I just implemented your solution, the test passed.